pkg/controller/node/nodecontroller.go

```diff
@@ -991,12 +1000,19 @@ func (nc *NodeController) tryUpdateNodeStatus(node *v1.Node) (time.Duration, v1.
 	return gracePeriod, observedReadyCondition, currentReadyCondition, err
 }
 
-func (nc *NodeController) checkForNodeAddedDeleted(nodes []*v1.Node) (added, deleted []*v1.Node) {
+func (nc *NodeController) checkForNodeAddedDeleted(nodes []*v1.Node) (added, deleted, updated []*v1.Node) {
```

Comment:
I don't like wording. 'updated' accompanied by 'added' and 'deleted' suggests that it means updated Nodes, not new zones. Please rename it to sth like 'newZoneRepresentatives' and add a comment explaining return values.